### PR TITLE
Do not refresh model when search query changes

### PR DESCRIPTION
**Pull request by balajimanoharan**
- Do not refresh model when search query changes
- Update search bar styles
- Add clear search icon

newsearch

### CODE REVIEW

File: packages/directory/app/styles/navi-directory/components/dir-search-bar.less [hunk 1]
```diff
@@ -5,26 +5,35 @@
...
     align-items: center;
     display: grid;
-    grid-template-columns: 580px 20px;
-    grid-template-areas: "search-input  search-icon";
-    margin-left: 10px;
-
-    &__input:focus {
-        border-bottom-color: rgba(25, 143, 255, 0.75);
-    }
+    grid-template-columns: 20px 1fr 40px;
```
Comment (mattgraham1995) [inline packages/directory/app/styles/navi-directory/components/dir-search-bar.less]: 🔤

File: packages/directory/tests/unit/mixins/search-filter-test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import EmberObject from '@ember/object';
+import SearchFilterMixin from 'navi-directory/mixins/search-filter';
+import { module, test } from 'qunit';
+import { set } from '@ember/object';
+import { A as arr } from '@ember/array';
+
+module('Unit | Mixin | search-filter', function() {
+  // Replace this with your real tests.
```
Comment (mattgraham1995) [inline packages/directory/tests/unit/mixins/search-filter-test.js]: Extraneous comment

File: packages/directory/addon/templates/components/dir-search-bar.hbs [hunk 3]
```diff
@@ -5,4 +5,8 @@
     class='dir-search-bar__input'
     key-up=(action this.attrs.searchFor)
 }}
-{{navi-icon 'search' class='dir-search-bar__search-icon'}}
\ No newline at end of file
+{{navi-icon 'search' class='dir-search-bar__search-icon'}}
+{{navi-icon 'times' 
+    class=(concat 'dir-search-bar__clear-icon' (if (not-eq query.length 0) ' dir-search-bar__clear-icon--show'))
+    click=(action (mut query) '')
```
Comment (mattgraham1995) [inline packages/directory/addon/templates/components/dir-search-bar.hbs]: Should this `clear search` functionality be tested?

Comment (balajimanoharan) [reply packages/directory/addon/templates/components/dir-search-bar.hbs]: done

File: packages/directory/app/styles/navi-directory/components/dir-search-bar.less (hunk 1 above)
Comment (balajimanoharan) [reply packages/directory/app/styles/navi-directory/components/dir-search-bar.less]: done

File: packages/directory/addon/mixins/search-filter.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    directory: controller(),
+
+    /**
+     * @property searchResults - Search and rank through items in model when a search query is available
+     */
+    searchResults: computed('directory.q', 'model', function() {
+        let items = get(this, 'model'),
+            queryString = get(this, 'directory.q');
+
+        return queryString && queryString.length ? searchRecords(items, queryString, 'title') : items;
```
Comment (Xuefeng-Zhu) [inline packages/directory/addon/mixins/search-filter.js]: Can we use isEmpty here

**Pull request merged**